packages/did-client/src/did-client.ts

```diff
@@ -1,90 +1,308 @@
+        //     verificationMethod: [
+        //         {
+        //           id: 'did:ethr:0x61:0x599b3912A63c98dC774eF3E60282fBdf14cda748#controller',
+        //           type: 'EcdsaSecp256k1RecoveryMethod2020',
+        //           controller: 'did:ethr:0x61:0x599b3912A63c98dC774eF3E60282fBdf14cda748',
+        //           blockchainAccountId: '0x599b3912A63c98dC774eF3E60282fBdf14cda748@eip155:97'
+        //         },
+        //         {
+        //           id: 'did:ethr:0x61:0x599b3912A63c98dC774eF3E60282fBdf14cda748?context=0x678904eb5c3f53d8506e7085dfbb0ef333c5f7d0769bcaf4ca2dc0ca46d00001',
+        //           type: 'EcdsaSecp256k1VerificationKey2019',
+        //           controller: 'did:ethr:0x61:0x599b3912A63c98dC774eF3E60282fBdf14cda748',
+        //           publicKeyHex: '12345bb792710e80b7605fe4ac680eb7f070ffadcca31aeb0312df80f7300001'
+        //         },
+        //         {
+        //           id: 'did:ethr:0x61:0x599b3912A63c98dC774eF3E60282fBdf14cda748?context=0x67890c45e3ad1ba47c69f266d6c49c589b9d70de837e318c78ff43c7f0b00003',
+        //           type: 'Ed25519VerificationKey2018',
+        //           controller: 'did:ethr:0x61:0x599b3912A63c98dC774eF3E60282fBdf14cda748',
+        //           publicKeyBase58: '2E4cfzc9Kf2nvScMZ2bJwGKPn19TJYvPE98D8RCguqL6'
+        //         }
+        //     ],
+        //     assertionMethod: [
+        //         'did:ethr:0x61:0x599b3912A63c98dC774eF3E60282fBdf14cda748#controller',
+        //         'did:ethr:0x61:0x599b3912A63c98dC774eF3E60282fBdf14cda748?context=0x678904eb5c3f53d8506e7085dfbb0ef333c5f7d0769bcaf4ca2dc0ca46d00001',
+        //         'did:ethr:0x61:0x599b3912A63c98dC774eF3E60282fBdf14cda748?context=0x67890c45e3ad1ba47c69f266d6c49c589b9d70de837e318c78ff43c7f0b00003'
+        //     ],
+        //     authentication: [
+        //         'did:ethr:0x61:0x599b3912A63c98dC774eF3E60282fBdf14cda748#controller',
+        //         'did:ethr:0x61:0x599b3912A63c98dC774eF3E60282fBdf14cda748?context=0x678904eb5c3f53d8506e7085dfbb0ef333c5f7d0769bcaf4ca2dc0ca46d00001'
+        //     ],
+        //     service: [
+        //         {
+        //             id: 'did:ethr:0x61:0x599b3912A63c98dC774eF3E60282fBdf14cda748?context=0x84e5fb4eb5c3f53d8506e7085dfbb0ef333c5f7d0769bcaf4ca2dc0ca4698fd4&type=message',
+        //             type: 'VeridaMessage',
+        //             serviceEndpoint: 'https://db.testnet.verida.io:5002'
+        //         },
+        //         {
+        //             id: 'did:ethr:0x61:0x599b3912A63c98dC774eF3E60282fBdf14cda748?context=0xcfbf4621af64386c92c0badd0aa3ae3877a6ea6e298dfa54aa6b1ebe00769b28&type=database',
+        //             type: 'VeridaDatabase',
+        //             serviceEndpoint: 'https://db.testnet.verida.io:5002'
+        //         }
+        //     ]
+        //   }
     }
 
+    public authenticate(privateKey: string) {
+        throw new Error('Method not implemented.')
```

Comment:
This method accepts a private key and then unlocks the ability to save documents.

It should be possible to "get" a document, without providing a private key.

Reply:
Once privateKey not provided in constructor, there should be an identifier in config. If not we can't create a vda-did-resolver and consequently not able to get document.

Reply:
Implemented.